**<p style="font-size: 35px; text-align: center">Probability Distributions</p>**



***<center>Miguel Ángel Vélez Guerra</center>***

<hr/>

![Distribuciones](https://4.bp.blogspot.com/-ImwjGBnN9Yg/VuYgbbaNBJI/AAAAAAAAA_o/rdXnY7x6I8svIEsXRcm51-jrj_Lopdb-w/s1600/E2-U3.png)


<hr/>
<hr/>

**<p id="tocheading">Tabla de contenido</p>**
<br/>
<div id="toc"></div>


In [69]:
%%javascript
// Script to generate table of contents
$.getScript('../resources/table_of_contents.js')

<IPython.core.display.Javascript object>

<hr/>
<hr/>

## Imports

In [1]:
#-------Importing from other folder------#

import sys
sys.path.insert(0, "../resources/")

import mstats as ms

#-----------Miguel's statistics----------#

In [2]:
import scipy.stats as ss
import numpy as np

<hr/>
<hr/>

## 1. Distribución binomial
De acuerdo con el periódico de educación superior (Journal of
Higher Educatión), el 40% de todos los bachilleres trabajan durante el verano
para ganar dinero para pagar la educación universitaria. Si 7 estudiantes se
seleccionan de forma aleatoria, ¿Cuál es la probabilidad de que 1) 5 tengan
trabajos en verano, 2) ninguno trabaje, 3) todos trabajen?, 4) ¿Cuál es la probabilidad de que 3 o menos
estudiantes trabajen? 5) ¿Cuál es la probabilidad de que 5 o más
estudiantes trabajen?

In [3]:
n_bachilleres = 7 # Tamaño de la muestra
pi_bachilleres = 0.4 # Probabilidad de éxito

**1) Probabilidad de que exactamente 5 tengan trabajos en verano (x = 5)**

In [4]:
x_bachilleres = 5 # Valor a determinar la probabilidad

In [5]:
ms.distributions.binom_pmf(n_bachilleres, pi_bachilleres, x_bachilleres)

0.07741440000000008

La probabilidad de que **exactamente** 5 estudiantes tengan trabajo en verano es de **0.07741440000000008**

**2) Probabilidad de que ningún estudiante tenga trabajos en verano (x = 0)**

In [6]:
x_bachilleres = 0

In [7]:
ms.distributions.binom_pmf(n_bachilleres, pi_bachilleres, x_bachilleres)

0.027993599999999987

La probabilidad de que ningún estudiante tenga trabajo en verano es de **0.027993599999999987**

**3) Probabilidad de que todos los estudiantes seleccionados tengan trabajos en verano (x = 7)**

In [8]:
x_bachilleres = 7

In [9]:
ms.distributions.binom_pmf(n_bachilleres, pi_bachilleres, x_bachilleres)

0.001638400000000001

La probabilidad de que todos los estudiantes tengan trabajo en verano es de **0.001638400000000001**

**4) ¿Cuál es la probabilidad de que 3 o menos estudiantes trabajen? (x <= 3)**

In [10]:
x_bachilleres = 3

In [11]:
ms.distributions.binom_cdf(n_bachilleres, pi_bachilleres, x_bachilleres)

0.710208

La probabilidad de que 3 o menos estudiantes trabajen es de **0.710208**

**5) ¿Cuál es la probabilidad de que 5 o más estudiantes trabajen? (x >= 5)**

In [12]:
x_bachilleres = 5

In [13]:
1 - ms.distributions.binom_cdf(n_bachilleres, pi_bachilleres, x_bachilleres)

0.018841600000000014

La probabilidad de que 5 o más estudiantes trabajen es de **0.018841600000000014**

## 2. Distribución hipergeométrica
Supongamos que en un establo de carreras hay N = 10 caballos y r = 4 de ellos
tienen una enfermedad contagiosa. ¿Cuál es la probabilidad de seleccionar
una muestra de n = 3 caballos, en la cual x = 2 caballos estén enfermos?

In [14]:
N_caballos = 10 # Tamaño de la población
r_caballos = 4 # Número de casos especiales en la población
n_caballos = 3 # Tamaño de la muestra

**1) ¿Cuál es la probabilidad de seleccionar una muestra de n = 3 caballos, en la cual x = 2 caballos estén enfermos?**

In [15]:
x_caballos = 2 # Número de casos especiales en la muestra

In [16]:
ms.distributions.hypergeom_pmf(N_caballos, r_caballos, n_caballos, x_caballos)

0.30000000000000016

La probabilidad de que 2 caballos estén enfermos es de **0.30000000000000016**

<hr/>
<hr/>

## 3. Distribución hipergeométrica
En un caso reciente en el distrito de Jhonson en Kansas City, tres
mujeres entablaron una demanda contra una empresa de servicios locales,
por discriminación de sexos. De las 9 personas que eran elegibles para un
ascenso, cuatro eran mujeres. Tres de las nueve personas recibieron en
realidad el ascenso; pero solo una de ellas era mujer. Las otras tres mujeres
elegibles demandaron. Una consideración importante en el caso, unida con la
probabilidad de que de las tres personas que recibieron ascenso sólo una
mujer fuera seleccionada por casualidad. Es decir, si el género no era un
factor, ¿Cuál es la probabilidad de que no más que uno de los tres ascensos
fuera asignado a una mujer?

In [17]:
N_discriminacion = 9 # Tamaño de la población
r_discriminacion = 4 # Número de casos especiales en la población
n_discriminacion = 3 # Tamaño de la muestra

**¿Cuál es la probabilidad de que no más que uno de los tres ascensos fuera asignado a una mujer? (x <= 1)**

In [18]:
x_discriminacion = 1 # Número de casos especiales en la muestra

In [19]:
ms.distributions.hypergeom_cdf(N_discriminacion, r_discriminacion, n_discriminacion, x_discriminacion)

0.5952380952380948

La probabilidad de que no más que uno de los tres ascensos fuera asignado a una mujer es del **0.5952380952380948**. 

Por lo que se puede concluir que no hay discriminación de genero, pues hay mayores posibilidades de que se ascienda a una mujer que a un hombre.

<hr/>
<hr/>

## 4. Distribución Poisson
Supongamos que se está interesado en la probabilidad de que exactamente 5
clientes lleguen durante la siguiente hora (o en cualquier hora dada) laboral.
La observación simple de las últimas 80 horas ha demostrado que 800
clientes han entrado al negocio. Por tanto, μ = 10 por hora.

In [20]:
mu_clientes = 800/80 # Número promedio de ocurrencias de un evento

**La probabilidad de que exactamente 5 clientes lleguen durante la siguiente hora (o en cualquier hora dada) laboral (x = 5)**

In [21]:
x_clientes = 5 # Número de veces que ocurre el evento

In [22]:
ms.distributions.poisson_pmf(mu_clientes, x_clientes)

0.03783327480207079

La probabilidad de que exactamente 5 clientes lleguen en un intervalo de tiempo de 1 hora es de **0.03783327480207079**

<hr/>
<hr/>

## 5. Distribución Poisson
El profesor Bradley anima a sus estudiantes de estadística a “actuar de forma
prudente” consultando al tutor si tienen alguna pregunta mientras se preparan para
el examen final. Parece que la llegada de los estudiantes a la oficina del tutor se
ajusta a una distribución de Poisson, con un promedio de 5.2 estudiantes cada 20
minutos. El profesor Bradley está preocupado porque si muchos estudiantes
necesitan los servicios del tutor, puede resultar un problema de congestión.

a. El tutor debe determinar la probabilidad de que cuatro estudiantes lleguen
durante cualquier intervalo de 20 minutos, lo cual podría causar el problema de 
Uso de Probabilidades para el Análisis y la Gestión
congestión que teme el profesor Bradley. Si la probabilidad excede el 20%, se
contratará un segundo tutor.

b. El tutor debe calcular la probabilidad de que más de cuatro estudiantes lleguen
durante algún período de 20 minutos. Si es mayor que el 50%, las horas de
oficina del tutor se aumentarán, permitiendo a los estudiantes extender el
horario en las que vienen a ver al tutor.

c. Si la probabilidad de que más de siete estudiantes lleguen durante un período
cualquiera de 30 minutos excede 50%, el mismo profesor Bradley ofrecerá
tutoría adicional.


In [23]:
mu_estudiantes = 5.2 # Número promedio de ocurrencias de un evento

**a) Determinar la probabilidad de que cuatro estudiantes lleguen durante cualquier intervalo de 20 minutos (x = 4)**

In [24]:
x_estudiantes = 4 # Número de veces que ocurre el evento

In [25]:
ms.distributions.poisson_pmf(mu_estudiantes, x_estudiantes)

0.16806250284943824

La probabilidad de que lleguen 4 estudiantes en un intervalo de 20 minutos a la oficina es de **0.16806250284943824**.

El profesor Bradley no debería contratar un segundo tutor, pues esta probabilidad no excede el *20%*.

**b) Calcular la probabilidad de que más de cuatro estudiantes lleguen durante algún período de 20 minutos (x > 4)**

In [26]:
x_estudiantes = 4

In [27]:
1 - ms.distributions.poisson_cdf(mu_estudiantes, x_estudiantes)

0.5938719984274373

La probabilidad de que más de 4 estudiantes lleguen en un intervalo de tiempo de 20 minutos es de **0.5938719984274373**.

Por lo que las horas de oficina del tutor deben aumentarse, pues esta probabilidad excede el *50%*

**c) La probabilidad de que más de siete estudiantes lleguen durante un período cualquiera de 30 minutos (x > 7)**

In [28]:
mu_estudiantes = 5.2/20 * 30
x_estudiantes = 7

In [29]:
1 - ms.distributions.poisson_cdf(mu_estudiantes, x_estudiantes)

0.5187909829079025

La probabilidad de que más de 7 estudiantes lleguen en un intervalo de tiempo de 30 minutos es de **0.5187909829079025**.

Entonces el profesor Bradley ofrecerá tutoría adicional pues la probabilidad excede el *50%*.

<hr/>
<hr/>

## 5. Distribución exponencial
Se asume que la tasa promedio de llegada de los clientes es de μ = 1.5 por hora y se desea saber la probabilidad de que no más de 2 horas transcurran entre llamadas.

In [30]:
mu_llamadas = 1.5 # Número promedio de ocurrencias de un evento

**La probabilidad de que no más de 2 horas transcurran entre llamadas. (x <= 2)**

In [31]:
x_llamadas = 2 # Lapso de tiempo entre ocurrencias de un evento

In [32]:
ms.distributions.expon_cdf(mu_llamadas, x_llamadas)

0.950212931632136

La probabilidad de que no transcurran más de 2 horas entre llamadas es de **0.950212931632136**

<hr/>
<hr/>

## 6. Distribución exponencial
Cross City Cab Company programan sus taxis para que lleguen al
aeropueto local en una distribución Poisson con una tasa promedio de
llegada de 12 por hora. Usted acaba de aterrizar en el aeropuerto y debe
llegar al centro a cerrar un gran negocio. ¿Cuál es la probabilidad de que
usted tenga que esperar máximo 5 minutos para conseguir un taxi? Su jefe es
un tirano que no tolerará la falla, de manera que si la probabilidad de que
pase otro taxi dentro de 5 minutos es menor al 50%, usted alquilará un carro
para el viaje a la oficina.

In [33]:
mu_taxis = 12/60 # Número promedio de ocurrencias de un evento

**¿Cuál es la probabilidad de que usted tenga que esperar máximo 5 minutos para conseguir un taxi? (x <= 5)**

In [34]:
x_taxis = 5 # Lapso de tiempo entre la ocurrencia del evento

In [35]:
ms.distributions.expon_cdf(mu_taxis, x_taxis)

0.6321205588285577

La probabilidad de que se tengan que esperar como máximo 5 minutos para conseguir un taxi es de **0.6321205588285577**.

Por lo que usted no tendrá que alquilar un caro para el viaje a la oficina.

<hr/>
<hr/>

## 7. Distribución uniforme
Suponga que los contenidos de las latas de 16 onzas de fruta enlatada producida por Del Monte oscila entre 14.5 y 17.5 onzas y que se ajusta a una distribución uniforme.

Del monte, desea saber la probabilidad de que una sola lata pese entre 16 y 17.2 onzas.

In [36]:
a_monte = 14.5 # Valor mínimo de la distribución
b_monte = 17.5 # Valor máximo de la distribución

**La probabilidad de que una sola lata pese entre 16 y 17.2 onzas. (16 <= x <= 17.2)**

In [37]:
x1_monte = 16 # Valor mínimo de la evaluación
x2_monte = 17.2 # Valor máximo de la evaluación

In [38]:
ms.distributions.uniform_cdf(a_monte, b_monte, x2_monte) - ms.distributions.uniform_cdf(a_monte, b_monte, x1_monte)

0.3999999999999998

La probabilidad de que una sola lata pese entre 16 y 17.2 onzas es de **0.3999999999999998**

<hr/>
<hr/>

## 8. Distribución uniforme
Dow Chemical produce un fertilizante inorgánico para césped,
para aquellos que fertilizan su pasto en casa, de manera que lo puedan podar
con más frecuencia. Este tipo de fertilizante se vende en bolsas con un peso
uniformemente distribuido, con media de 25 libras y un rango de 2.4 libras.
Harry Homeowner necesita 23 libras para fertilizar su césped, pero duda si
comprar sólo una bolsa ya que se desvía de las 25 libras en un rango de 2.4
libras. También tiene curiosidad sobre la probabilidad de comprar una bolsa
con más de 25.5 libras.

In [39]:
a_dow = 25 - 1.2 # Valor mínimo de la distribución
b_dow = 25 + 1.2 # Valor máximo de la distribución

**Probabilidad de que pese entre 21.8 y 24.2 libras (21.8 <= x <= 24.2)**

In [40]:
x1_dow = 23 - 1.2 # Valor mínimo de la evaluación
x2_dow = 23 + 1.2 # Valor máximo de la evaluación

In [41]:
ms.distributions.uniform_cdf(a_dow, b_dow, x2_dow) - ms.distributions.uniform_cdf(a_dow, b_dow, x1_dow)

0.16666666666666619

La probabilidad de que Harry consiga una bolsa de 23 libras es de **0.16666666666666619**.

**Probabilidad de que pese más de 25.5 libras**

In [42]:
x_dow = 25.5

In [43]:
1 - ms.distributions.uniform_cdf(a_dow, b_dow, x_dow)

0.2916666666666665

La probabilidad de comprar una bolsa que pese más de 25.5 libras es de **0.2916666666666665**.

<hr/>
<hr/>

## 9. Distribución normal
Topps Wear, un gran fabricante de ropas, desea estudiar la distribución en la estatura de
las personas. Topps Wear reconoció que el público estaba en constante cambio en su
tamaño físico y en sus proporciones. Se supone que si Topps Wear fuera a medir las
estaturas de todos sus clientes potenciales, encontraría que las estaturas están
distribuidas normalmente alrededor de una media de 67 pulgadas. Se asume que la
desviación estándar de los clientes es de 2 pulgadas.

El personal de Topps Wear puede hallar la probabilidad de que un solo cliente tenga entre
67 y 69 pulgadas de estatura, P(67 ≤ X ≤ 69), simplemente hallando el área que está
bajo la curva normal entre 67 y 69. Si se conoce el área, se conocerá la probabilidad.

In [44]:
mu_topps = 67 # Valor promedio
sigma_topps = 2 # Desviación estándar

**La probabilidad de que un solo cliente tenga entre 67 y 69 pulgadas de estatura, (67 ≤ X ≤ 69)**

In [45]:
x1_topps = 67
x2_topps = 69

In [46]:
ms.distributions.norm_cdf(mu_topps, sigma_topps, x2_topps) - ms.distributions.norm_cdf(mu_topps, sigma_topps, x1_topps)

0.3413447460685429

La probabilidad de que un solo cliente tenga entre 67 y 69 pulgadas de estatura es de **0.3413447460685429**

<hr/>
<hr/>

## 10. Aproximación normal a la distribución binomial
Se considera un sindicato laboral en el cual el 40% de los miembros está a favor de una
huelga. Si se seleccionan 15 miembros de manera aleatoria, ¿Cuál es la probabilidad de que
10 apoyen un paro?

In [47]:
n_sindicato = 15 # Tamaño de la muestra
pi_sindicato = 0.4 # Probabilidad de éxito

**¿Cuál es la probabilidad de que 10 apoyen un paro? (x = 10)**

In [48]:
x_sindicato = 10

In [49]:
ms.distributions.binom_pmf(n_sindicato, pi_sindicato, x_sindicato)

0.02448564210892806

La probabilidad de que 10 apoyen un paro es de **0.02448564210892806**.

<hr/>
<hr/>

## 11. Distribuciones muestrales
TelCom Satellite presta servicios de comunicación a los negocios del área
metropolitana de Chicago. Los funcionarios de la compañía han aprendido
que la transmisión de satélite promedio es de 150 segundos, con una
desviación estándar de 15 segundos. Los tiempos parecen estar distribuidos
normalmente.
TelCom desea ahora conocer la probabilidad de que la media de n = 50
llamadas esté entre 150 y 155 segundos.

In [50]:
mu_telcom = 150 # Valor promedio
sigma_telcom = 15 # Desviación estándar
n_telcom = 50 # Tamaño de la muestra

**La probabilidad de que la media de n = 50 llamadas esté entre 150 y 155 segundos. (150 <= x <= 155)**

In [51]:
x1_telcom = 150
x2_telcom = 155

In [52]:
ms.distributions.sample_norm_cdf(mu_telcom, sigma_telcom, n_telcom, x2_telcom) - ms.distributions.sample_norm_cdf(mu_telcom,
                                                                                                                 sigma_telcom,
                                                                                                                 n_telcom,
                                                                                                                 x1_telcom)

0.4907889372729505

La probabilidad de que la media de 50 llamadas esté entre 150 y 155 segundos es de **0.4907889372729505**.

<hr/>
<hr/>

## 12. Proporciones muestrales
BelLabs adquiere componentes para sus teléfonos celulares en
lotes de 200 de una firma en Palo Alto. El componente tiene una tasa de
defectos del 10%. Una política establecida recientemente por BelLabs
establece que si el siguiente envío tiene:
1. Más del 12% de defectos, definitivamente buscará otro proveedor.
2. Entre el 10 y el 12% de defectos, considerará un nuevo proveedor.
3. Entre el 5 y el 10% de defectos, definitivamente no conseguirá un nuevo
proveedor.
4. Menos del 5% de defectos, incrementará sus pedidos.
¿Cuál es la decisión más probable que tome BelLabs?

In [60]:
n_bellabs = 200 # Tamaño de la muestra
pi_bellabs = 0.10 # Probabilidad de éxito

In [63]:
1 - ms.distributions.sample_prop_cdf(n_bellabs, pi_bellabs, 0.12)

0.17288929307558032

In [64]:
ms.distributions.sample_prop_cdf(n_bellabs, pi_bellabs, 0.12) - ms.distributions.sample_prop_cdf(n_bellabs, pi_bellabs, 0.10)

0.3271107069244197

In [66]:
ms.distributions.sample_prop_cdf(n_bellabs, pi_bellabs, 0.10) - ms.distributions.sample_prop_cdf(n_bellabs, pi_bellabs, 0.05)

0.4907889372729505

In [67]:
ms.distributions.sample_prop_cdf(n_bellabs, pi_bellabs, 0.05)

0.009211062727049501